trees/src/iavl/node_db.rs

```diff
@@ -76,20 +76,20 @@ where
         [NODES_PREFIX.to_vec(), hash.to_vec()].concat()
     }
 
-    pub(crate) fn get_node(&self, hash: &[u8; 32]) -> Option<Node> {
+    pub(crate) fn get_node(&self, hash: &[u8; 32]) -> Option<Box<Node>> {
         let cache = &mut self.cache.lock().expect("Lock will not be poisoned");
         let cache_node = cache.get(hash);
 
         if cache_node.is_some() {
-            return cache_node.map(|v| v.to_owned());
+            return cache_node.map(|v| Box::new(v.to_owned()));
```

Comment:
Maybe it's better to cache boxed nodes?

Reply:
I agree, they should probably be on the heap. I'll create a separate issue for this.

Reply:
Actually hash maps store their data on the heap so boxing doesn't change much in this regard. Was there some other reason for wanting to store boxed values?

Reply:
You're right, but there is an issue with `Box`. When you do `Box::new(T)` T value is first allocated on stack then cloned onto heap.

Reply:
The get method of the cache returns a borrowed value, so we would still need a way to copy that borrowed value into a new box.

Reply:
Okay. Let's leave it as is for now.